debian/base-entrypoint.sh

```diff
@@ -1,15 +1,14 @@
 #!/bin/sh
 
-# create alone user/group so we aren't forced to run app as root
-id -u alone 2>/dev/null
-if [ $? -ne 0 ]; then
-  groupadd -r alone --gid=$APP_GROUP_ID
-  useradd -r -g alone -d /home/alone -c "alone app user" --uid=$APP_USER_ID alone
-  echo "created alone user/group with UID/GID: $APP_USER_ID/$APP_GROUP_ID"
-else
-  echo "alone user already exists"
+CURRENT_UID=`id -u node 2>/dev/null`
+CURRENT_GID=`id -g node 2>/dev/null`
+
+# if either APP_USER_ID or APP_GROUP_ID environment variables exist,
+# the node user's uid/gid will be updated accordingly. this is useful
+# if you want to avoid permissions issues when mounting volumes
+if [ ! -z "$APP_USER_ID" -o ! -z "$APP_GROUP_ID" ]; then
+  groupmod -g ${APP_GROUP_ID:-$CURRENT_GID} node && \
+  usermod -u ${APP_USER_ID:-$CURRENT_UID} -g ${APP_GROUP_ID:-$CURRENT_GID} node
 fi
-chown alone:alone /home/alone
```

Comment:
Don't we still need these?  The kernel won't update UIDs of files when `usermod` updates `/etc/passwd`, so when you change the UID `node`, all the files will still be owned by the old UID.

Reply:
oh really?  i pulled this from another image i was using, i thought i had tested that...let me go take a second look

Reply:
lol, well i learned something today https://github.com/moby/moby/issues/6119  that's quite a thread.  are you on a new enough version of the docker engine to use the `--chown` flag on COPY?  if not, then `chown -R`'s will be needed.  although i would recommend putting that responsibility on the child image's `docker-entrypoint.sh`, since `chown -R` sucks for directories with a large amount of files.

Reply:
i think we're better off keeping any `chown`'s and `chmod`'s out of the base image, defaulting to the more secure approach of keeping everything root.  the child image/entrypoint can take care of any permission updates that might be needed.

Reply:
OK -- we're breaking the "contract" with docker-node users/dependent images, but that change makes sense.